File: backend/app/api/audit.py
"""Account-wide audit log API — spec section 52 (Phase 12)."""
from fastapi import APIRouter, Depends, Query
from sqlalchemy.ext.asyncio import AsyncSession

from app.api.deps import get_current_user
from app.database.session import get_db
from app.models.user import User
from app.schemas.audit import AuditLogRead
from app.services.audit_service import list_audit_log

router = APIRouter(prefix="/audit-log", tags=["audit"])


@router.get("", response_model=list[AuditLogRead])
async def get_audit_log(
    limit: int = Query(default=100, ge=1, le=500),
    user: User = Depends(get_current_user),
    db: AsyncSession = Depends(get_db),
):
    return await list_audit_log(db, user.id, limit=limit)

File: frontend/src/services/audit.ts
import { api } from "./api";
import type { AuditLogEntry } from "@/types/audit";

export async function getAuditLog(limit = 100): Promise<AuditLogEntry[]> {
  const { data } = await api.get<AuditLogEntry[]>("/audit-log", { params: { limit } });
  return data;
}
